[READY] Plumbing improvements and chemical synthesizers

Time-Green opened the pull request:
Removes dumb visual bugs.
Put the copy pasted wrench stuff for disabling and enabling the component, in the component using a new signal from default_unfasten_wrench.
Add a general plumbing machinery subtype. Right now it's mainly for cataloging.
Adds the chemical synthesizer. You can select a single chemical and an amount between 0 and 5. It will produce it every tick. Should be combined with other synthesizers to create meth factories. 
Makes the liquid pump a fully functioning machine.

:cl:
add: The station comes one step closer to plumbing with the introduction of the chemical synthesizer.
sprite: sprites for the synthesizer, input and output commissioned by @Meyhazah
/:cl:

I wanted to do all the plumbing basic equipment at once, but stuff like plumbing heaters is gonna be ugly if I don't rework them completely and I don't wanna scare the maintainers away. 

There is slight copypasting from the chem_dispenser and with the chem synthesizer, but having it be inhereted would be a worse clusterfuck IMO.

Time-Green committed:
Finishes pump and fixes visual bug with disconnecting plumbing objects

Time-Green committed:
More bugfixes, finishes liquid pump

Time-Green committed:
Merge remote-tracking branch 'upstream/master' into fluids-2-electric-boogaloo

Time-Green committed:
Chemical synthesizer, wrench signal to handle unwrenching

nemvar commented on code/modules/plumbing/plumbers/synthesizer.dm:
```diff
@@ -0,0 +1,104 @@
+	///The maximum we can produce for every process
+	var/max_amount = 4
+	///I track them here because I have no idea how I'd make tgui loop like that
+	var/static/list/possible_amounts = list(1,2,3,4)
```
0 should probably be a possible amount, as it doesn't look like this has an off switch.

Time-Green replied:
Good idea

ATH1909 commented on code/modules/plumbing/plumbers/synthesizer.dm:
```diff
@@ -0,0 +1,104 @@
+	///The maximum we can produce for every process
+	var/max_amount = 4
+	///I track them here because I have no idea how I'd make tgui loop like that
+	var/static/list/possible_amounts = list(1,2,3,4)
```
shouldn't "5" be in this list?

Time-Green replied:
Ducts cant transfer more than 5 units at a time, so it would make mixing impossible. I guess there’s no harm in doing it anyway, in-case someone just wants retarted amounts of one chemical

Time-Green committed:
Merge remote-tracking branch 'upstream/master' into fluids-2-electric-boogaloo

Time-Green committed:
adds input and output pipes and changes ob into obj

Time-Green committed:
dmdocs all of plumbing through proper means

Time-Green committed:
adds sprites by meyhaza

also added wrenching interaction and fixed a dumb dumb with me fucking up an object path

Time-Green committed:
brightens the sprites a little by me

optimumtact commented:
I am trying to walk towards a world where chemicals aren't just synthesised out of thin air but actually collected in game, so Im quite hesitant to add more synthesis sources.

optimumtact commented:
what is your end goal with them?

Time-Green commented:
I want to add full fledged chemical factories. People would be able to combine this with the unsynthesizable exotic stuff from geysers to do all kinds of advanced automated processes. Or make meth factories for all the addicts.

Imagine pumping special lava oil, processing it and then selling it for money. Or connecting it to a pill press for pills.

Alternatively I could add a special reagent, atom fluid or something, that needs to be consumed by chem dispensers for them to synthesize stuff, so it doesn’t create matter from thin air.

optimumtact commented:
that's still basically the same thing, but your spacechem idea does sounds interesting.

Time-Green committed:
Merge remote-tracking branch 'upstream/master' into fluids-2-electric-boogaloo

Time-Green committed:
removes big select from synthesizer ui

optimumtact commented:
You know, this in miniature was basically my exact plan for chemistry anyway.

:thinking: :bulb:

Time-Green committed:
Merge remote-tracking branch 'upstream/master' into fluids-2-electric-boogaloo

Time-Green committed:
update tgui.js thing

Time-Green commented:
Conflict is no more

Time-Green committed:
unfuck pump color

apparently the replace all of spriting has downsides

Time-Green committed:
cleans up ui

actioninja commented:
tgui conflict

Time-Green committed:
Merge remote-tracking branch 'upstream/master' into fluids-2-electric-boogaloo

Time-Green commented:
conflict is gone once more
